packages/core/src/Engine.ts

```diff
@@ -537,6 +551,35 @@ export class Engine extends EventDispatcher {
           camera._cameraIndex = index;
         }
       );
+
+      const uiCanvases = componentsManager._overlayCanvases._elements;
+      if (uiCanvases) {
+        const {
+          _canvas: canvas,
+          _uiRenderQueue: uiRenderQueue,
+          _virtualCamera: virtualCamera,
+          _renderContext: renderContext,
+          _batcherManager: batcherManager
+        } = this;
+        const { elements: projectE } = virtualCamera.projectionMatrix;
+        const { elements: viewE } = virtualCamera.viewMatrix;
+        (projectE[0] = 2 / canvas.width), (projectE[5] = 2 / canvas.height), (projectE[10] = 0);
```

Comment:
To reduce redundant access to virtualCamera properties and improve readability, I suggest destructuring projectionMatrix, viewMatrix, and viewProjectionMatrix at the beginning and directly working with elements throughout the code. Additionally, this helps minimize repetitive operations in the loop. Here’s an optimized version:

const { projectionMatrix, viewMatrix, viewProjectionMatrix } = virtualCamera;
const projectE = projectionMatrix.elements;
const viewE = viewMatrix.elements;

// Update matrices and perform operations...